pkg/router/template/configmanager/haproxy/manager.go

```diff
@@ -199,6 +204,14 @@ func (cm *haproxyConfigManager) AddBlueprint(route *routeapi.Route) {
 	newRoute.Namespace = blueprintRoutePoolNamespace
 	newRoute.Spec.Host = ""
 
+	if cm.extendedValidation {
+		if err := validateBlueprintRoute(newRoute); err != nil {
+			glog.Errorf("Skipping blueprint route %s/%s due to invalid configuration: %v",
```

Comment:
If invalid routes are present, what's the impact to the cluster, and what action must the admin take? What's the likelihood the admin would ever see this error? Are invalid blueprint route errors surfaced in some other way (e.g. via status)?

Reply:
So in the normal case if a route is invalid, the route is _not_ admitted by the router and the status is updated (if router has perms) with the invalidation errors. 

In the case of blueprint routes, we aren't really updating the status - it may well end up happening (explained below) but we can only log here that there was an error and ignore that blueprint.

Now its possible the blueprint route is in a namespace that is also being served by the router - in which case the router will update the status of the blueprint route (well as far as the router is concerned in this case its just another route) with the errors.

Reply:
Will the error continue to happen every resync interval as long as the naughty route exists? Would an event be more useful?

Reply:
Hmm, that's true - yeah, it would happen every resync interval.

Reply:
@ironcladlou  looked at event generation fixes here - it might end up being a wee bit too invasive here for a bug fix as we don't have the client here, etc.
I'd rather do that separately in a follow on PR - which you can decide whether or not to put in this release. That sound good?

Reply:
Had a chance to actually read some of the controller code.... looks like the admin controls the namespace containing the route templates, and so probably also controls the routes in the blueprint namespace. So, if there's an invalid route there which was causing an issue of any kind, the route could presumably be modified to be valid or be removed. Does that sound accurate? Thanks for being patient with explanations!